compositor_api/src/types/from_register_input.rs

```diff
@@ -102,13 +102,19 @@ impl TryFrom<RtpInput> for pipeline::RegisterInputOptions {
         }
 
         let rtp_stream = input::rtp::RtpStream {
-            video: video.as_ref().map(|video| input::rtp::InputVideoStream {
-                options: match video {
-                    InputRtpVideoOptions::FfmepgH264 => decoder::VideoDecoderOptions {
-                        codec: pipeline::VideoCodec::H264,
-                    },
-                },
-            }),
+            video: video
+                .as_ref()
+                .map(|video| {
+                    Ok(input::rtp::InputVideoStream {
+                        options: match video {
+                            InputRtpVideoOptions::FfmepgH264 => decoder::VideoDecoderOptions {
+                                codec: pipeline::VideoCodec::H264,
+                                decoder: pipeline::VideoDecoder::FFmpegH264,
```

Comment:
do we need a separate field for codec and decoder?

Reply:
yup, because we have multiple decoders for a single codec

Reply:
and ffmpeg can decode all videos, while the hardware decoder is limited by our implementation and the underlying hardware, e.g. some GPUs might only be able to decode the `baseline` profile, while others may be able to decode `high`